src/package.cc

```diff
@@ -45,12 +52,16 @@ double Package::GetFillMass(double qty) {
   return fill_mass;
 }
   
-Package::Package() : id_(next_package_id_++), fill_min_(0), fill_max_(std::numeric_limits<double>::max()) {
-  name_ = "unnamed";
-}
-
-Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : name_(name), id_(next_package_id_++), fill_min_(fill_min), fill_max_(fill_max), strategy_(strategy) {}
-
-Package::Package(int id, std::string name) : id_(id), name_(name), fill_min_(0), fill_max_(std::numeric_limits<double>::max()) {}
+Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : 
+  name_(name), fill_min_(fill_min), fill_max_(fill_max), strategy_(strategy) {
+    if (name == unpackaged_name_) {
+      if (unpackaged_) {
+        throw ValueError("can't create a package with name " + unpackaged_name_);
+      }
+      id_ = unpackaged_id_;
```

Comment:
I'm not sure I've covered all the bases here... it's possible for the user to generate a package with this reserved name before the unpackaged object gets built, which will lead to problems.